framework/src/systems/NonlinearSystemBase.C

```diff
@@ -1137,8 +1138,12 @@ NonlinearSystemBase::setConstraintSecondaryValues(NumericVector<Number> & soluti
             // reinit variables on the primary element's face at the contact point
             _fe_problem.setNeighborSubdomainID(primary_elem, 0);
             subproblem.reinitNeighborPhys(primary_elem, primary_side, points, 0);
+
+            std::vector<Point> reference_points;
+            FEInterface::inverse_map(
+                primary_elem->dim(), FEType(), primary_elem, points, reference_points);
             _fe_problem.reinitNeighborPhys(
-                _mesh.elemPtr(primary_elem->id()), primary_side, points, 0);
+                _mesh.elemPtr(primary_elem->id()), primary_side, reference_points, 0);
```

Comment:
You won't be able to use the reference points directly in this call because `reinitNeighborPhys` expects physical points. Unless you use a different API you'll need to forward map the reference points to their physical points on the undisplaced mesh